src/python/pants/backend/python/lint/black/rules.py

```diff
@@ -149,7 +149,7 @@ def generate_args(
     return Setup(process_request)
 
 
-@rule(name="Format using Black")
+@named_rule(name="black_fmt", desc="Format using Black", rule_type=NamedRuleType("format"))
```

Comment:
I think I'll remove the notion of a `rule_type` from this PR then. We don't actually need it yet, and when we do need it it might be better to just compute the JSON entry from structural information about the graph like you suggest, rather than requiring an additional annotation.